src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -49,14 +50,22 @@ const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValida
   const value = useValue(uischema);
   const { loading } = useWidgetContext();
   const onChangeHandler = useOnChange(uischema);
-  const label = getTranslation(uischema.translations!, 'label');
   const { translations = [], focus, required } = uischema;
   const {
     attributes,
     inputMeta: { name },
   } = uischema.options;
+  const label = getTranslation(translations, 'label');
+  const hint = getTranslation(translations, 'hint');
+  const explain = getTranslation(translations, 'bottomExplain');
   const focusRef = useAutoFocus<HTMLInputElement>(focus);
   const hasErrors = typeof errors !== 'undefined';
+  const hintEleId = `${name}-hint`;
+  const explainEleId = `${name}-explain`;
+  const hintAndExplainIds = `${explain ? ` ${explainEleId} ` : ''}${hint ? ` ${hintEleId}` : ''}`;
```

Comment:
Probably easier to read if you do the ternary logic separately. And this will result in something like `password-hintpassword-explain`; do you want to add a delimiter/separator between the first/second half?

```suggestion
  const hintEleId = hint ? `${name}-hint` : '';
  const explainEleId = explain ? `${name}-explain` : '';
  const hintAndExplainIds = `${explainEleId}${hintEleId}`;
```